lib/core/data-types/Factory.js

```diff
@@ -2,6 +2,9 @@ import MantraDataType from './MantraDataType';
 import { MantraString, MantraBoolean, MantraNumber } from './index';
 import { error } from '../../utils';
 
+/*
+    This object contains all the supported MantraDataTypes.
+*/
 const DATA_TYPES = {
```

Comment:
Let's take this opportunity to move the DATA_TYPES const serving as 'core of our factory' to the create() method and pass it by params to the method needed it. Or even it could be a get () on the class (your pick ;))

This is in order to avoid circular dependencies by moving these constant from being a module property and let then be evaluated in a 'non-sync' execution for the module resolving such as closured in a method or obtained by a class getter.

Reply:
Sure! Will do.

I will pick the "method" approach to be consistency because the MantraData already uses a method to obtain all of the MantraComponent classes in order to avoid any circular dependencies.